src/Microsoft.ML.ImageAnalytics/ImageLoader.cs

```diff
@@ -201,11 +239,112 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
+                Contracts.Assert(0 <= iinfo && iinfo < _parent.ColumnPairs.Length);
+
+                disposer = null;
+                var getSrc = input.GetGetter<ReadOnlyMemory<char>>(input.Schema[ColMapNewToOld[iinfo]]);
+                ReadOnlyMemory<char> src = default;
+                ValueGetter<VBuffer<byte>> del =
+                    (ref VBuffer<byte> dst) =>
+                    {
+                        getSrc(ref src);
+
+                        if (src.Length > 0)
+                        {
+                            string path = src.ToString();
+                            if (!string.IsNullOrWhiteSpace(_parent.ImageFolder))
+                                path = Path.Combine(_parent.ImageFolder, path);
+
+                            int imgSize = LoadDataIntoBuffer(path, ref dst);
+                            if (imgSize < 0)
+                                throw Host.Except($"Failed to load image {src.ToString()}.");
+                        }
+                    };
+
                 return del;
             }
 
+            public static int LoadDataIntoBuffer(string path, ref VBuffer<byte> imgData)
+            {
+                int count = -1;
+                int bytesread = -1;
+                // bufferSize == 1 used to avoid unnecessary buffer in FileStream
+                using (FileStream fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1))
+                {
+                    long fileLength = fs.Length;
+                    if (fileLength > int.MaxValue)
+                        throw new IOException($"File {path} too big to open.");
+                    else if (fileLength == 0)
+                    {
+                        Console.WriteLine("File length is zero even though the image is not empty");
+                        byte[] imageBuffer;
+                        // Some file systems (e.g. procfs on Linux) return 0 for length even when there's content.
+                        // Thus we need to assume 0 doesn't mean empty.
+                        imageBuffer = File.ReadAllBytes(path);
+                        count = imageBuffer.Length;
+                        imgData = new VBuffer<byte>(count, imageBuffer);
+                        return count;
+                    }
+
+                    count = (int)fileLength;
+                    var editor = VBufferEditor.Create(ref imgData, count);
+
+#if NETSTANDARD2_0
+                    bytesread = ReadToEnd(fs, editor.Values);
+                    Contracts.Assert(count == bytesread);
+
+#else
+                    fs.Read(editor.Values);
+                    bytesread = editor.Values.Length;
+                    Contracts.Assert(count == bytesread);
+#endif
+                    imgData = editor.Commit();
+                    return count;
+
+                }
+
+            }
+
+            public static int ReadToEnd(System.IO.Stream stream, Span<byte> bufferspan)
```

Comment:
I think this is only called from `#if NETSTANDARD2_0`, so it can be wrapped in `#if NETSTANDARD2_0` itself, so we don't use it when we aren't on netstandard2.0.

Reply:
Changing it. Thanks.